packages/common/src/model/ModelTransformFunctions.ts

```diff
@@ -416,9 +417,20 @@ const doUpload = async (projectName, fileName, buffer) => {
 
 const toProjectAndFileName = (fUploadPath: string, srcBaseURL: string): [string, string] => {
   // TODO: remove srcBaseURL if it's unnecessary
-  const [_, projectName, fileName] =
-    MATCH_ASSET_PROJECT_FILENAME_REGEX.exec(fUploadPath) ??
-    MATCH_ASSET_PROJECT_FILENAME_REGEX.exec(pathJoin(srcBaseURL, fUploadPath))!
+  let matchResult: RegExpExecArray | null
+  if (srcBaseURL.includes('publish')) {
```

Comment:
let's get a more stringent conditional to determine if this is a publish url. Either that or make a separate function for the publish workflow